### PR TITLE
Implementation of CRSF v3

### PR DESCRIPTION
**Pull request by andyp1per**
Full implementation of CRSFv3 including baud rate negotiation, new link frame statistics and new variable bit rate RC frames

Flown on a CRSFv2 copter to check for regressions
Flown on tracer CRSFv3 - works fine

### CODE REVIEW

**Comment by Hwurzburg** [comment]
can we get an accessor for RX link quality...for both CRSF v2 and v3? for OSD support

**Comment by andyp1per** [comment]
Code is now correct, but baud rate change step causes the RC to lose connection.

**Review by yaapu — APPROVED** [review]

File: libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp [hunk 1]
```diff
@@ -355,6 +463,22 @@ void AP_RCProtocol_CRSF::process_link_stats_frame(const void* data)
     _link_status.rf_mode = static_cast<RFMode>(MIN(link->rf_mode, 3U));
 }
 
+// process link statistics to get RX RSSI
+void AP_RCProtocol_CRSF::process_link_stats_rx_frame(const void* data)
+{
+    const LinkStatisticsRXFrame* link = (const LinkStatisticsRXFrame*)data;
+
+    _link_status.rssi = link->rssi_percent * 255.0f / 100.0f;
```
Comment (tridge) [inline libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp:471]: * 0.01

File: libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp [hunk 2]
```diff
@@ -355,6 +463,22 @@ void AP_RCProtocol_CRSF::process_link_stats_frame(const void* data)
...
+}
+
+// process link statistics to get TX RSSI
+void AP_RCProtocol_CRSF::process_link_stats_tx_frame(const void* data)
+{
+    const LinkStatisticsTXFrame* link = (const LinkStatisticsTXFrame*)data;
+
+    _link_status.rssi = link->rssi_percent * 255.0f / 100.0f;
```
Comment (tridge) [inline libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp:479]: * 0.01

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp:483]: Fixed

File: libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp:491]: Fixed

**Comment by andyp1per** [comment]
Now flown with fw 6.07 - all working fine

**Review by tridge — APPROVED** [review]

**Pull request merged**